src/libstd/keyword_docs.rs

```diff
@@ -1578,12 +1578,181 @@ mod type_keyword {}
+/// extract from [`Vec::set_len`]. The `# Safety` section explains the contract
+/// that must be fulfilled to safely call the function. If necessary, a
+/// `# Panics` section can be added to document possible panics and their
+/// reasons.
+///
+/// ```rust,ignore (stub-to-show-doc-example)
+/// /// Forces the length of the vector to `new_len`.
+/// ///
+/// /// This is a low-level operation that maintains none of the normal
+/// /// invariants of the type. Normally changing the length of a vector
+/// /// is done using one of the safe operations instead, such as
+/// /// `truncate`, `resize`, `extend`, or `clear`.
+/// ///
+/// /// # Safety
+/// ///
+/// /// - `new_len` must be less than or equal to `capacity()`.
+/// /// - The elements at `old_len..new_len` must be initialized.
+/// ///
+/// pub unsafe fn set_len(&mut self, new_len: usize)
+/// ```
+///
+/// # Using `unsafe` blocks and `impl`s
+///
+/// Calling an `unsafe` function requires an `unsafe {}` block, which can take
+/// two forms. Note that the `enclosing_fn` form is the subject of [proposal]
+/// to modify its behavior as an `unsafe {}` block.
+///
+/// ```rust
+/// # #![allow(dead_code)]
+/// unsafe fn unsafe_fn() {}
+///
+/// // This function is unsafe to call and unsafe functions can be called
+/// // inside without problems.
+/// unsafe fn enclosing_unsafe() {
+///     unsafe_fn()
+/// }
+///
+/// // This function is safe to call. Any call to an unsafe function inside
+/// // must be enclosed in an unsafe block. `interior_unsafe` is responsible
+/// // for ensuring that memory unsafety never occurs, no matter how or when
+/// // it is called.
+/// fn interior_unsafe() {
+///     // SAFETY: `unsafe_fn` does nothing.
+///     unsafe { unsafe_fn() }
+/// }
+/// ```
+///
+/// Traits marked as `unsafe` must be [`impl`]emented using `unsafe`. This makes
+/// a guarantee to other `unsafe` code that a property holds for this type. It
+/// is undefined behavior to make an `unsafe impl` for a type which does not
+/// have this property. The [Send] and [Sync] are examples of traits that are
+/// `unsafe` to implement.
+///
+/// ```rust
+/// # #![allow(dead_code)]
+/// unsafe trait UnsafeTrait {}
+///
+/// struct S;
+///
+/// // SAFETY: explanation about why this impl is safe.
+/// unsafe impl UnsafeTrait for S {}
+/// ```
+///
+/// See the [Rustnomicon] and the [Reference] for more informations.
+///
+/// [`extern`]: keyword.extern.html
+/// [`trait`]: keyword.trait.html
+/// [`static`]: keyword.static.html
+/// [`union`]: keyword.union.html
+/// [`impl`]: keyword.impl.html
+/// [Send]: marker/trait.Send.html
+/// [Sync]: marker/trait.Sync.html
+/// [`Vec::set_len`]: vec/struct.Vec.html#method.set_len
+/// [raw pointers]: ../reference/types/pointer.html
 /// [memory safety]: ../book/ch19-01-unsafe-rust.html
-/// [not yet complete]: https://github.com/rust-lang/rust/issues/34601
+/// [Rustnomicon]: ../nomicon/index.html
+/// [Nomicon]: ../nomicon/safe-unsafe-meaning.html
```

Comment:
Some of these link names are not very descriptive, maybe introduce some aliases? "Nomicon" and "Rustnomicon" not pointing to the same thing is surprising, for example.

You can introduce aliases as follows:
```
soundness in the [Nomicon][nomicon-soundness]

[nomicon-soundness]: https://...
```

Reply:
Done !